samples/bidirectional-chat/csharp/Function.cs

```diff
@@ -0,0 +1,170 @@
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Azure.WebJobs.Extensions.SignalRService;
+
+namespace bidirectional_chat
+{
+    public static class Function
+    {
+        private const string Hub = "simplechat";
+        private const string NewMessageTarget = "newMessage";
+        private const string NewConnectionTarget = "newConnection";
+
+        [FunctionName("negotiate")]
+        public static SignalRConnectionInfo GetSignalRInfo(
+            [HttpTrigger(AuthorizationLevel.Anonymous)] HttpRequest req,
+            [SignalRConnectionInfo(HubName = Hub, UserId = "{headers.x-ms-signalr-user-id}")] SignalRConnectionInfo connectionInfo)
+        {
+            return connectionInfo;
+        }
+
+        [FunctionName("connect")]
+        public static async Task Connect([SignalRTrigger(Hub, Category.Connections, Event.Connect)]InvocationContext invocationContext,
+            [SignalR(HubName = Hub)]IAsyncCollector<SignalRMessage> signalRMessages)
+        {
+            await signalRMessages.AddAsync(new SignalRMessage()
+            {
+                Target = NewConnectionTarget,
+                Arguments = new object[] { new NewConnection(invocationContext.ConnectionId) }
+            });
+        }
+
+        [FunctionAuthorize]
+        [FunctionName("broadcast")]
+        public static async Task Broadcast([SignalRTrigger(Hub, Category.Messages, "broadcast", "message")]InvocationContext invocationContext,
```

Comment:
>broadcast [](start = 84, length = 9)

How about enum messages event as well. the functionality is limited to our supported APIs.